skimage/transform/_thin_plate_splines.py

```diff
@@ -0,0 +1,299 @@
+import numpy as np
+import scipy as sp
+
+
+class TPSTransform:
+    def __init__(self):
+        self._estimated =  False
+        self.parameters = np.array([], dtype=np.float32)
+        self.control_points = np.array([], dtype=np.float32)
+
+
+    def __call__(self, points):
+        """Map the source points to the destination surface.
+
+        Parameters
+        ----------
+        new_coords : array_like
+            Array of source points to be transformed
+        Returns
+        -------
+            ndarray : Mapped point of the distination point
+        """
+        x_src = _ensure_2d(points)
+
+
+        if x_src.shape[1] != self.control_points.shape[1]:
+            raise ValueError("Array must be identical")
+
+        dist = self._radial_distance(x_src)
+        transforms = np.hstack([dist, np.ones((x_src.shape[0], 1)), x_src])
+        return transforms @ self.parameters
+
+    @property
+    def inverse(self):
+        raise NotImplementedError("This is yet to be implemented.")
+
+    def estimate(self, dst, src):
+        """Estimate how close is the deformed source to the target.
+
+        Number of source and destination points must match.
+
+        Parameters
+        ----------
+        src : (N, 2) array_like
+            Control point at source coordinates
+        dst : (N, 2) array_like
+            Control point at destination coordinates
+
+        Returns
+        -------
+        success: bool
+            True, if all pieces of the model are successfully estimated.
+        """
+
+        src = _ensure_2d(src)
+        dst = _ensure_2d(dst)
+
+        if src.shape != dst.shape:
+            raise ValueError("src and dst shape must be identical")
+
+        # if src.shape[-1] != 2 and dst.shape[-1] != 2:
+        #     raise ValueError("src and dst must have shape (N,2)")
+
+        self.control_points = src
+        n , d = src.shape
+
+        K = self._radial_distance(src)
+        P = np.hstack([np.ones((n, 1)), src])
+        O = np.zeros((3, 3))
+        L = np.asarray(np.bmat([[K, P],[P.T, O]]))
+        Y = np.concatenate([dst, np.zeros((d + 1, d))])
+        self.parameters = np.dot(np.linalg.pinv(L), Y)
+
+        self._estimated = True
+        return self._estimated
+
+
+    def _transform_points(self, x, y, coeffs):
+        w = coeffs[:-3]
+        a1, ax, ay, = coeffs[-3:]
+        summation = np.zeros(x.shape)
+        for wi, Pi in zip(w, self.control_points):
+            r = np.sqrt((Pi[0] - x)**2 + (Pi[1] - y)**2)
+            summation += wi * _U(r)
+        return a1 + ax*x + ay*y + summation
+
+    def transform(self, x, y):
```

Comment:
I am confused by this method. From it's docstring and behavior it seems to do the same thing as the API
```python
tform = TPSTransform()
tform.estimate(...)
tform(...)
```
is supposed to achieve? Do you follow?

Reply:
Yes, I follow. My recent commit applies this suggestions.